pstoremem/addr_book.go

```diff
@@ -52,14 +63,18 @@ type memoryAddrBook struct {
 }
 
 var _ pstore.AddrBook = (*memoryAddrBook)(nil)
+var _ pstore.CertifiedAddrBook = (*memoryAddrBook)(nil)
 
-func NewAddrBook() pstore.AddrBook {
+func NewAddrBook() *memoryAddrBook {
 	ctx, cancel := context.WithCancel(context.Background())
 
 	ab := &memoryAddrBook{
 		segments: func() (ret addrSegments) {
 			for i, _ := range ret {
-				ret[i] = &addrSegment{addrs: make(map[peer.ID]map[string]*expiringAddr)}
+				ret[i] = &addrSegment{
+					addrs:             make(map[peer.ID]map[string]*expiringAddr),
+					signedAddrs:       make(map[peer.ID]map[string]*expiringAddr),
+					signedPeerRecords: make(map[peer.ID]*peerRecordState)}
```

Comment:
How about?

* unsigned
* signed
* records

Reply:
Actually, do we need to store two maps for signed and unsigned? IUC, all we have is one set of mutually exclusive addresses.

If we have signed addresses, we can't have unsigned ones, and vice versa. Would it be sufficient to store one addrs map, and a boolean indicating whether they're signed or not?

That would enable us to take the decisions we need to evaluate during manipulation, right?